### PR TITLE
Delete comments by ID

### PR DESCRIPTION
**Pull request by prashantiA**
New Features
- Added individual button for each comment to delete individually

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java [hunk 1]
```diff
@@ -20,9 +20,30 @@
...
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    deleteAll();
+    String type = request.getParameter("type");
+    if (type == null) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      System.err.println("No parameter for type");
+    }
+    else if (type.equals(DELETE_ALL)) deleteAll();
```
Comment (lee12jin) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java:33]: https://google.github.io/styleguide/javaguide.html#s4.1-braces
Please use braces. Secondly, please overlap any else if and else statements.

if () {
} else if () {
} else {
}

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java [hunk 2]
```diff
@@ -20,9 +20,30 @@
...
+    else if (type.equals(BY_ID)) {
+      try {
+        long id = Long.parseLong(request.getParameter("id"));
+	deleteById(id);
```
Comment (jasperlee-google) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java:37]: nit: think there needs to be a few spaces before `deleteById(id);`.

(similar to `response.setStatus(HttpServletResponse.SC_BAD_REQUEST);`)

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -81,19 +81,31 @@ function formatComment(comment) {
...
+  const response = await fetch('/delete-comments?type=all', {method : 'post'});
   
   /* Timeout selected experimentally to prevent most issues with latency
    * There are still some latency issues but too much lag was introduced by making the timeout longer
    */
   setTimeout(loadComments, 1000);
 }
 
+async function deleteOneComment(id) {
+  const response = await fetch('delete-comments?type=byId&id='+id, {method : 'post'});
```
Comment (lee12jin) [inline portfolio/src/main/webapp/script.js:104]: Add the "/"

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java (hunk 1 above)
Comment (jasperlee-google) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java:33]: Can you add braces for the one line `else if` statement?

Context:
https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java (hunk 2 above)
Comment (prashantiA) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java]: Fixed the indentation issue, thanks for catching this!

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java (hunk 1 above)
Comment (prashantiA) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java]: Fixed the braces on the if else statements

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (prashantiA) [reply portfolio/src/main/webapp/script.js]: Added the "/", thanks for catching this!

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java (hunk 1 above)
Comment (prashantiA) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java]: Fixed the braces issues

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java [hunk 4]
```diff
@@ -20,9 +20,30 @@
...
+      } catch (NumberFormatException e) {
+        System.err.println("Failed to parse to long: " + request.getParameter("id"));
+	response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
```
[the remark this answers is not in the document]
Comment (prashantiA) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java]: Fixed indentation

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (prashantiA) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java]: Fixed indentation

**Pull request merged**